### PR TITLE
Extended Malaria Infohub Posts to support Images and External Links

### PR DESCRIPTION
**Pull request by yatna**
**MALARIA HUB POST**

Create Post - 

screen shot 2017-06-24 at 7 57 22 pm

View Post-

screen shot 2017-06-24 at 8 11 52 pm

Edit Post-

screen shot 2017-06-24 at 8 12 05 pm

### CODE REVIEW

**Comment by yatna** [comment]
**FIRST AIDE - GET HELP NOW POSTS**

Create Post

screen shot 2017-06-25 at 7 25 54 pm

View Post -

screen shot 2017-06-25 at 7 36 26 pm


Edit post -

screen shot 2017-06-25 at 7 36 37 pm

File: malaria_web/templates/malaria/create_post.html [hunk 1]
```diff
@@ -50,6 +50,36 @@ <h2 class="text-center inverted-font">New Post</h2>
...
+                            {% endfor %}
+                        {% endif %}
+                    </strong>
+                </p>
+            </div>
+        </div>
+        <div class="form-group">
+            <label class="col-sm-2 control-label inverted-font">Photo</label>
+            <div class="col-sm-8">
+                <input type="file"  name="photo" accept=".jpg,.png,.jpeg">
```
Comment (medhach) [inline malaria_web/templates/malaria/create_post.html]: Maybe say 'image' instead of 'photo'

**Review by BeiyuLinCS — APPROVED** [review]

File: malaria_web/models.py [hunk 2]
```diff
@@ -20,7 +22,12 @@ class Post(models.Model):
                                             )]
                                         )
     # link to important documents
-    link_post = models.CharField(max_length=2000)
+    link_post = models.CharField(max_length=200)
```
Comment (medhach) [inline malaria_web/models.py:25]: Why have you changed it from 2000 to 200?

File: malaria_web/services.py [hunk 3]
```diff
@@ -15,11 +15,11 @@ def create_post_from_form(form, owner):
     return post
 
 
-def create_revpost(owner, post, title, description):
+def create_revpost(owner, post, title, description, link, test):
```
Comment (medhach) [inline malaria_web/services.py]: By test, you mean the image? If yes, change the name so its more readable :)

File: pcsa_GHN/models.py [hunk 4]
```diff
@@ -1,11 +1,12 @@
 from django.db import models
 
 from signup.models import Pcuser
+from django.core.files.storage import FileSystemStorage
 
 
 # Create your models here.
 class Contact(models.Model):
-    office_name = models.CharField(max_length=200)
+    office_name = models.CharField(max_length=190)
```
Comment (medhach) [inline pcsa_GHN/models.py]: Why 190 from 200 exactly?

File: malaria_web/models.py (hunk 2 above)
Comment (yatna) [reply malaria_web/models.py:25]: Generally URLs are short, so I kept the max_length to 200.

File: malaria_web/services.py (hunk 3 above)
Comment (yatna) [reply malaria_web/services.py]: Okay, I will

File: pcsa_GHN/models.py (hunk 4 above)
Comment (yatna) [reply pcsa_GHN/models.py]: Umm.. I guess I was just checking by changing the models if migrations were occuring or not, I will revert it back.

**Comment by amittbhardwj** [comment]
@yatna you can use UrlField instead for links.

On Jul 11, 2017 2:01 PM, "Yatna Verma" <notifications@github.com> wrote:

*@yatna* commented on this pull request.
------------------------------

In malaria_web/models.py
<https://github.com/systers/macc/pull/308#discussion_r126764248>:

> @@ -20,7 +22,12 @@ class Post(models.Model):
                                             )]
                                         )
     # link to important documents
-    link_post = models.CharField(max_length=2000)
+    link_post = models.CharField(max_length=200)

Generally URLs are short, so I kept the max_length to 200.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/systers/macc/pull/308#discussion_r126764248>, or mute
the thread

**Pull request merged**